downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/GenerateGOAnnotationFile/CellularComponentAnnotationBuilder.java

```diff
@@ -0,0 +1,84 @@
+                    } else {
+                        logger.info("Species has an alternative GO compartment, skipping GO annotation");
+                    }
+                } else {
+                    logger.info("Protein is from an excluded microbial species, skipping GO annotation");
+                }
+            } else {
+                logger.info("Invalid protein, skipping GO annotation");
+            }
+        }
+    }
+
+    /**
+     * Proteins reaching this part of the script are deemed valid for GO annotation. Retrieves the Cellular Compartment accession associated with this instance and then calls the GOA line generator.
+     * @param proteinInst -- GKInstance, Individual protein instance from the retrieved proteins.
+     * @param referenceEntityInst -- GKInstance, Reference Entity instance of the protein.
+     * @param reactionInst -- GKInstance, Parent reaction that the protein is associated with.
+     * @param taxonIdentifier -- String, CrossReference ID of protein's species.
+     * @throws Exception -- MySQLAdaptor exception.
+     */
+    private static void generateGOCellularCompartmentLine(GKInstance proteinInst, GKInstance referenceEntityInst, GKInstance reactionInst, String taxonIdentifier) throws Exception {
+        String reactomeIdentifier = GOAGeneratorConstants.REACTOME_IDENTIFIER_PREFIX  + ((GKInstance) reactionInst.getAttributeValue(ReactomeJavaConstants.stableIdentifier)).getAttributeValue(ReactomeJavaConstants.identifier).toString();
+        String goCellularCompartmentAccession = getCellularCompartmentGOAccession(proteinInst);
+        if (!goCellularCompartmentAccession.isEmpty()) {
+            String goaLine = GOAGeneratorUtilities.generateGOALine(referenceEntityInst, GOAGeneratorConstants.CELLULAR_COMPONENT_LETTER, goCellularCompartmentAccession, reactomeIdentifier,  GOAGeneratorConstants.TRACEABLE_AUTHOR_STATEMENT_CODE, taxonIdentifier);
+            GOAGeneratorUtilities.assignDateForGOALine(proteinInst, goaLine);
+        } else {
+            logger.info("Protein has no Cellular Compartment accession, skipping GO annotation");
+        }
+    }
+
+    /**
+     * Checks for and returns valid GO accession specific to Cellular Compartments in the protein of interest.
+     * @param proteinInst -- GKInstance, Individual protein instance from the retrieved proteins.
+     * @return -- String, GO Cellular Compartment accession from proteinInst.
+     * @throws Exception -- MySQLAdaptor exception.
+     */
+    private static String getCellularCompartmentGOAccession(GKInstance proteinInst) throws Exception {
+        GKInstance compartmentInst = (GKInstance) proteinInst.getAttributeValue(ReactomeJavaConstants.compartment);
+        String cellularComponentAccession = "";
+        if (compartmentInst != null && !GOAGeneratorUtilities.proteinBindingAnnotation(compartmentInst.getAttributeValue(ReactomeJavaConstants.accession))) {
```

Comment:
Like above, a `doesNotHaveProteinBindingAnnotation` method would be good here.  Maybe included in the GOAGeneratorUtilities class since it is used here and by BiologicalProcess